### PR TITLE
Add support for Vroom 1.11.0

### LINKED ISSUE #24: Add support for Vroom 1.11
A new version of Vroom was released: https://github.com/VROOM-Project/vroom/releases/tag/v1.11.0

At this stage of the project I would say that we don't need to consider backwards compatibility in case there are relevant changes.

### PR DESCRIPTION
**Pull request by krashish8**
Fixes #24

Changes proposed in this pull request:
- Added caching for VROOM in GitHub actions
- Pump up VROOM version to 1.11.0
  - Add setup time in jobs and shipments to refine service time modeling
  - Add support for custom cost matrices, along with the duration matrix.
    - Using `start_id`, `end_id`, `duration`, `cost` as matrix table columns.
  - Add `timeout` and `exploration_level` parameters to vroom-category functions.
  - Add `max_tasks` column in vehicles.
  - Add tests for empty skills arrays.
  - Add custom scaling logic for `speed_factor`
- Add Gehring & Homberger benchmark data for 1000 jobs: `c1_10_1` (Not testing in CI, takes lot of time)
- Honor client cancel requests for vroom-category functions.
- Modify parameter names: index->id and internal changes of start_time -> tw_open, end_time -> tw_close.
- Refined the documentation:
  - Separate INTEGER and TIMESTAMP/INTERVAL.
  - Add default values for all optional columns.
  - Update VROOM installation instructions, add the link in README
  - Update vrpRouting documentation link
- Update release_notes and NEWS.

@pgRouting/admins

### CODE REVIEW

**Review by krashish8 — COMMENTED** [review]
@cvvergara I suppose that it is okay to add this line because anything can change on `v0`. So, signature change should not be an issue, right?

Or do we have to release a v1 if any signature changes on v0?

File: tools/scripts/test_signatures.sh [hunk 1]
```diff
@@ -20,6 +20,9 @@ do
         # comparing within same mayors only
         if [ "$mayor1" != "$mayor2" ]; then continue; fi
 
+	# ignoring any signature changes made on v0
+	if [ "$mayor1" == 0 ]; then continue; fi
```
Comment (krashish8) [inline tools/scripts/test_signatures.sh:24]: This line

File: .github/workflows/boost_version.yml [hunk 2]
```diff
@@ -30,6 +30,7 @@ jobs:
           pgver=$(psql --version | grep -Po '(?<=psql \(PostgreSQL\) )[^;]+(?=\.\d \()')
           echo "PGVER=${pgver}" >> $GITHUB_ENV
           echo "PGIS=3" >> $GITHUB_ENV
+          echo "VROOMVER=1.11.0" >> $GITHUB_ENV
```
Comment (dkastl) [inline .github/workflows/boost_version.yml:33]: setting this as variable is already a very big improvement. I just saw that this is repeated for various workflows. I tried to find out, if there is a way to define an environment variable on a repository level for example, but couldn't find anything than secret variables.

**Review by dkastl — APPROVED** [review]
Great!

File: .github/workflows/boost_version.yml (hunk 2 above)
Comment (krashish8) [reply .github/workflows/boost_version.yml:33]: Yes, this can be done using secret variables. But it won't be a good way to do so, because whatever version we use in the workflow should not be a "secret". Normally, secret variables are supposed to be used to store some secret keys (environment variables), etc.

Comment (dkastl) [reply .github/workflows/boost_version.yml:33]: Yes, it's a bad idea to use secret variables. I was looking for something else ... a single place to set the VROOM version.
